src/GridGraph.flix

```diff
@@ -4,33 +4,41 @@ namespace Flixball/GridGraph {
     use Flixball/Core.Coordinates;
     use Flixball/Core.Direction;
     use Flixball/Core.Direction.{East, North, South, West};
+    use Flixball/Core.Move;
+    use Flixball/Core.Move.{Walk, Shoot, Turn};
+    use Flixball/Core.Position;
+    use Flixball/Core.Position.Position;
+    use Flixball/Core.Rotation;
+    use Flixball/Core.step;
     use Flixball/Core.Tile;
 
     ///
     /// It is assumed that `available` keeps the search space finite or
     /// that `goal` returns true for a reachable coordinate.
     /// This function might otherwise not halt.
+    /// The start coordinate will always be available no matter `available`.
     ///
     pub def findShortestPath(available: Coordinates -> Bool \ ef1,
-        start: {start :: Coordinates}, goal: Coordinates -> Bool \ ef2): Option[List[Coordinates]] \ {ef1, ef2} =
-    region r { if (goal(start.start)) Some(Nil) else {
+        start: Position, goal: Position -> Bool \ ef2): Option[List[Move]] \ {ef1, ef2} =
```

Comment:
Does the new Graph library help us at all?

Reply:
Possibly yeah. This function has 1) bail-out-fast when a goal is found and 2) a non-materialized graph, which we cannot have in `Graph`. Both of these are essentially a performance thing so maybe its not worth it for the complexity.

Reply:
If we end up with a 100x100 board then that could be important though, right?

Reply:
yeah, also since all ais probably do this each step